HW_2.2/hw_2.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+"""HW_2.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1wN8cVqAaHE6DE8tH12GF94qeOPonFXaZ
+
+# Задание 1 (2 балла)
+
+Напишите класс `MyDict`, который будет полностью повторять поведение обычного словаря, за исключением того, что при итерации мы должны получать и ключи, и значения.
+
+**Модули использовать нельзя**
+"""
+
+# Ваш код здесь
+class MyDict(dict):
+    def __iter__(self):
+        for key in super().__iter__():
+            yield key, self[key]
+
+dct = MyDict({"a": 1, "b": 2, "c": 3, "d": 25})
+for key, value in dct:
+    print(key, value)
+
+for key, value in dct.items():
+    print(key, value)
+
+for key in dct.keys():
+    print(key)
+
+dct["c"] + dct["d"]
+
+"""# Задание 2 (2 балла)
+
+Напишите функцию `iter_append`, которая "добавляет" новый элемент в конец итератора, возвращая итератор, который включает изначальные элементы и новый элемент. Итерироваться по итератору внутри функции нельзя, то есть вот такая штука не принимается
+```python
+def iter_append(iterator, item):
+    lst = list(iterator) + [item]
+    return iter(lst)
+```
+
+**Модули использовать нельзя**
+"""
+
+def iter_append(iterator, item):
+    for i in iterator:
+        yield i
```

Comment:
Так было бы то же самое, а то требуешь элемент из итератора, чтобы его же и перевыдать (вместо обоих строк 47-48).
```suggestion
        yield from i
```